### DISCUSSION TITLE
WindowBuilder Blank tab

### DISCUSSION THREAD

**Discussion by Poojav23** [eclipse/windowbuilder#178] [Q&A]
I am having a similar problem to #165 and #45. However, pasting old code into a new file and opening it with WindowBuilder did not fix the issue. Additionally, using the previous fix of adding "--illegal-access=permit" to the ini file is not applicable as that was already in it for me. Also re-installing the software did not help. I have attached a screenshot of the blank screen that happens both when opening a previous project w WindowBuilder and a new one. I am using JDK 17 with WindowBuilder 1.9.8. Any help would be greatly appreciated

![WindowBuilder Blank](https://user-images.githubusercontent.com/58189407/159066672-81a2539f-c5d1-4b34-842c-fd6feed92519.png)

**Comment by lcaron** [comment]
Sorry for this trouble. Can you please provide your OS, the JDK you use (Oracle, ...) and the version of Eclipse ?
Do you have any trace in the error log ?

**Comment by Poojav23** [comment]
I am using macOS Mojave(Version 10.14.6). I am using Oracle JDK 17.0.1. The Eclipse version is 4.22.0. Please let me know if you need any other info to troubleshoot this issue.

**Comment by wimjongman** [comment]
Please attach the file that you are trying to work with.

**Reply by wimjongman** [reply]
Also press CTRL+3 -> type 'copy configuration data to clipboard' -> paste in a text file, attach here (**do not paste in a comment**).

**Comment by Poojav23** [comment]
I have attached one of the many files that were created with windowbuilder, but are no longer allowing me to edit using the design tab.
[ViewClient.java.zip](https://github.com/eclipse/windowbuilder/files/8327679/ViewClient.java.zip)
Also here is the configuration data:
[configs.txt](https://github.com/eclipse/windowbuilder/files/8327702/configs.txt)

**Comment by lcaron** [comment]
Thank you for your feeback.
In your configuration file it seems that you use an older version of WB 
org.eclipse.wb.core (1.10.1.202106041225) "WindowBuilder Core" [Active]
org.eclipse.wb.core.databinding (1.9.1.202106041225) "WindowBuilder Databinding Core" [Starting]
org.eclipse.wb.core.java (1.9.2.202106041225) "WindowBuilder Core for Java GUI's" [Active]
org.eclipse.wb.core.lib (1.9.1.202106041225) "WindowBuilder Core Libraries" [Active]
org.eclipse.wb.core.ui (1.10.0.202106041225) "WindowBuilder Core UI" [Starting]
org.eclipse.wb.doc.user (1.9.1.202106041225) "WindowBuilder User Guide" [Starting]

Can you try to uninstall WB and re-install version 19.8 : https://download.eclipse.org/windowbuilder/1.9.8/  ?


Laurent

**Comment by Poojav23** [comment]
I already uninstalled and installed that version of windbowbuilder using the eclipse marketplace as shown in the attached screenshot:
![windowbuilder version](https://user-images.githubusercontent.com/58189407/159573475-d454bbbb-1771-447c-837d-11c8ae108114.png)
Would uninstalling it again and downloading from that link instead make a difference?

**Reply by wimjongman** [reply]
First uninstall WB and restart. Then make sure that all the wb plugins are removed from the configuration. As long as this is not the case, you have a corrupted installation.

For quick test, you can install Eclipse 4.23 fresh from https://eclipse.org/downloads and then install windowbuilder either from marketplace or from the update site ^^ and then see if the problem persists.

Your problems are 100% caused by the old wb plugins. They are not part of our 1.9.8 installation.

**Reply by wimjongman** [reply]
You can uninstall by using help/about -> installation details find the wb features and uninstall them.

**Reply by Poojav23** [reply]
This is a screenshot of my eclipse.ini file. I don't see any mention of windowbuilder is it called something else in the installation details? Also would just taking out those lines be enough to uninstall the plugin or should I additionally uninstall  it from within eclipse?
![Eclipse Ini](https://user-images.githubusercontent.com/58189407/159722483-fd649671-2cf3-4972-b3d1-2ca815807404.png)

**Reply by wimjongman** [reply]
You can look at the configuration with

help/about -> Installation Details -> Configuration Tab

**Reply by Poojav23** [reply]
I completely uninstalled it:
![uninstalled](https://user-images.githubusercontent.com/58189407/160034479-18083a15-b27c-4c33-a7c8-a12a4ea94f66.png)
 The windowbuilder details from the config file were still there however:
![configs](https://user-images.githubusercontent.com/58189407/160034709-ab2f40fb-1abf-4c27-9af3-901d16d36ae8.png)
Could this be the issue because when I reinstalled it using the url the bug was still there? If so how do I take this out of the configs(as opposed to uninstalling the software the traditional way) as the path from help/about -> Installation Details -> Configuration Tab does not allow for editing and just deleting the windowbuilder involved section.

**Comment by wimjongman** [comment]
You can remove the bundles from the plugins directory, the dropins directory, and the bundles.info file eclipse\configuration\org.eclipse.equinox.simpleconfigurator\bundles.info

Then restart with

     eclipse.exe -clean

Please note that this is unusual.

**Comment by GriffonTheorist697** [comment]
I don't know if my situation is the same as yours or [Woah Nelly](https://www.eclipse.org/forums/index.php/t/1093041/)'s, but I had this problem and I found a solution for it. While creating my class BoardWindow that extended JFrame, I attempted to pass in a parameter for a custom GameSimulator object I made. While updating WindowBuilder did not fix the issue, I did notice that WindowBuilder still worked fine if I created a new project. I then took out the parameter from the BoardWindow constructor and I made the GameSimulator into a singleton. After that, the WindowBuilder displayed my frame and components again!

Hope this helps!
